src/App.jsx

```diff
@@ -14,70 +16,95 @@ export const goods = [
-      Jam is selected
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState(START_SELECTED_GOOD);
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  const selectGood = (name) => {
+    const callback = () => setSelectedGood(name);
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+    return callback;
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  const unSelectGood = () => {
+    setSelectedGood('');
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  return (
+    <main className="section container">
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+      { selectedGood ? (
+        <h1 className="title is-flex is-align-items-center">
+          {`${selectedGood} is selected`}
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={unSelectGood}
+          />
+        </h1>
+      ) : (
+        <h1 className="title is-flex is-align-items-center">
+          No goods selected
+        </h1>
+      )
+      }
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+      <table className="table">
+        <tbody>
+          {
+            goods.map(
+              (good) => {
+                const isSelected = good === selectedGood;
+
+                let button = (
+                  <button
+                    data-cy="AddButton"
+                    type="button"
+                    className="button"
+                    onClick={selectGood(good)}
+                  >
+                    +
+                  </button>
+                );
+
+                if (isSelected) {
+                  button = (
+                    <button
+                      data-cy="RemoveButton"
+                      type="button"
+                      className="button is-info"
+                      onClick={unSelectGood}
+                    >
+                      -
+                    </button>
```

Comment:
instead of writing this button twice you can write:

```
 <button 
  data-cy={isSelected ? 'RemoveButton' : 'AddButton'}
  type="button"
  onClick={...}
  className={cn('button', {
    'is-info': isSelected,
  })}
>
  {isSelected ? '-' : '+'}
</button>
```